src/modules/order/order.dao.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@nestjs/common';
+import { Brackets, DeleteResult, Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import {
+  DEFAULT_PAGE_NO,
+  DEFAULT_PAGE_SIZE,
+} from '../../constants/DefaultPageParams';
+import { PageResponse } from '../../constants/PageResponse';
+import { Order } from '../../../entities/Order.entity';
+
+@Injectable()
+export class OrderDao {
+  constructor(
+    @InjectRepository(Order)
+    private readonly repository: Repository<Order>
+  ) {}
+
+  async saveOrder(order: Order): Promise<Order> {
+    return this.repository.save(order);
+  }
+
+  async getOrderById(orderId: string, companyId: string): Promise<Order> {
+    return this.repository
+      .createQueryBuilder('order')
+      .where('order.id = :orderId', { orderId })
+      .andWhere('order.company_id = :companyId', { companyId })
+      .getOne();
+  }
+
+  async findOrder(
+    companyId: string,
+    page = DEFAULT_PAGE_NO,
+    pageSize = DEFAULT_PAGE_SIZE,
+    searchCriteria: string,
+    order: 'ASC' | 'DESC',
+    orderBy: string
+  ): Promise<PageResponse<Order>> {
+    const query = this.repository
+      .createQueryBuilder('order')
+      .where('order.companyId = :companyId', { companyId });
+    if (searchCriteria) {
+      query.andWhere(
+        new Brackets(qb => {
+          // Update this code once the customer is merged to include the filter by customer name
```

Comment:
If this is the case add this as a requirement for @fernandogcabal when completing his Customer's feature

Reply:
I think this feature was already merged, let's check with @fernandogcabal to proceed and integrate the commented logic before merging

Reply:
@matheo97 an option is to include this in a new PR just to start merging code, but I can do that addition